src/Criteria.php

```diff
@@ -34,12 +34,16 @@ public function isEmpty(): bool
 
     public function has(string $name): bool
     {
-        return array_key_exists($name, $this->items);
+        return $this->get($name) !== null;
```

Comment:
I'm going to tell you something you told me last week: I think we're overcomplicating it for very little value :) Each Criterion MUST be unique in the collection by name. So to me, it makes perfect sense only to be able to check for its presence by that unique name. Why bother to check it by class name..?

Also, `has()` is ment to be a quick checkup method that doesn't need to resolve the value to check for its existence. That's why we index the name. Now, you actually need to try resolving the value in order to check if it exists.

I would release Distil without this modification tbh. If people really need this behaviour, we can still add it later on. But for now, it kinda messes up the simplicity of the collection.

Reply:
Ok, I agree